src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs

```diff
@@ -188,6 +191,47 @@ protected override void Write(HtmlRenderer renderer, CodeSnippet codeSnippet)
             renderer.Write("</code></pre>");
         }
 
+        private string GetNoteBookContent(string content, string tagName, CodeSnippet obj)
+        {
+            JObject contentObject = null;
+            try
+            {
+                contentObject = JObject.Parse(content);
+            }
+            catch (JsonReaderException ex)
+            {
+                _context.LogError("not-notebook-content", "Not a valid Notebook. " + ex.ToString(), obj);
+                return string.Empty;
+            }
+
+            string sourceJsonPath = string.Format("$..cells[?(@.metadata.name=='{0}')].source", tagName);
+            JToken sourceObject = null;
+            try
+            {
+                sourceObject = contentObject.SelectToken(sourceJsonPath);
+            }
+            catch (JsonException)
+            {
+                _context.LogError("mutiple-tags-with-same-name", string.Format("The name '{0}' is not unique in the notebook file.", tagName), obj);
+                return string.Empty;
+            }
+
+            if (sourceObject == null)
+            {
+                _context.LogError("source-code-notfound", string.Format("The name '{0}' is not present in the notebook file.", tagName), obj);
```

Comment:
`source-code-notfound` => `source-code-not-found` for consistency with existing codes

Reply:
As per my understanding, if source file is not found, source-code-not-found is used.  It is about tag not found. Changed accordingly.